File: backend/src/services/email.service.ts
export interface PasswordResetMessage {
  recipient: string;
  resetToken: string;
  expiresAt: Date;
}

export interface IEmailService {
  sendPasswordReset(message: PasswordResetMessage): Promise<void>;
}

export class ConsoleEmailService implements IEmailService {
  async sendPasswordReset(message: PasswordResetMessage): Promise<void> {
    if (process.env.NODE_ENV !== "production") {
      console.info(
        `[StudyConnect] Password reset for ${message.recipient}: token=${message.resetToken}, expires=${message.expiresAt.toISOString()}`
      );
    }
  }
}

export const emailService = new ConsoleEmailService();

File: backend-java/src/main/java/com/studyconnect/backend/service/EmailService.java
package com.studyconnect.backend.service;

import java.time.Instant;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;

@Service
public class EmailService {

    private static final Logger log = LoggerFactory.getLogger(EmailService.class);

    public void sendPasswordReset(String recipient, String resetToken, Instant expiresAt) {
        log.info("Password reset requested for {}. Token={} expiresAt={}", recipient, resetToken, expiresAt);
    }
}
